db/scripts/init_tables.sql

```diff
@@ -103,3 +103,4 @@ ALTER TABLE IF EXISTS aggregated_scans ADD COLUMN IF NOT EXISTS with_mxs INTEGER
 
 ALTER TABLE domains ADD COLUMN IF NOT EXISTS mta_sts BOOLEAN DEFAULT FALSE;
 
+ALTER TABLE aggregated_scans ADD UNIQUE (time, source);
```

Comment:
So I should have caught this on #234 (my bad!) but I think this will add an extra constraint metadata object onto the postgres table each time we do a migration. I did a thing with `domains_pkey` in the past where I drop & re-add the constraint, but that's also not a perfect solution since there's a brief period of time where the constraint doesn't exist!

Reply:
Interesting! How about a fancy fn like https://stackoverflow.com/a/6804058?

Reply:
I like your solution of putting them in a txn!

Reply:
Actually, I pushed a commit to wrap it in a transaction

Reply:
Okay, I updated it to use a real conditional. `domains_pkey` can still just use a transaction because `domains_pkey` will already exist on the first run, since it gets created when the domains table gets created.